OneCommunityGlobal/HighestGoodNetworkApp
# Miguel - Changing Feature Count To Zero Deletes Badge

# Description
Fixes a bug when setting the badge count to 0 (either manually or using arrows) and not deleting it afterward.
Now when a badge count reaches 0, a delete modal automatically pops up.  
Clicking on "Yes, Delete" will delete the badge "locally" but the "Save Changes" button needs to be pressed in the main badge modal to save the changes on the deleted badge in the server. 
If a user clicks on the "Cancel" button on the delete modal or if the user presses the "X" button on the main badge modal (without saving), the badges will revert back to the last saved badge counts.

Original Bug:
Login as administrator > ‘Welcome, Username’ > View Profile > Featured Badges > Assign Badges → Assign any quantity of  Badges to the user, then go to ‘Selected Featured’ and change the count to zero. The Badges should disappear when the ‘count’ is set to zero, but they don’t.

## Related PRS (if any):
n/a

## Main changes explained:
- Updated deletion logic for badges to account for both having a badge count of 0 and pressing the "Delete" button on a badge.
- Fixed sortBadges to not use a "shallow copy" of the server object causing server-sided changes badges to be updated "locally" without going through the proper methods.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. Go to your own profile or a separate profile (volunteer, manager, etc.)
5. Assign badges if none exist
6. Select Featured > set badge count to 0 (either manually typing it to 0 or using arrow keys)
7. Press Yes, Delete on Modal and Save Changes
8. verify the functionality of badge deletion by pressing "Select Featured" button and ensuring the deleted badges are not there.

Test with the featured checkbox being checked on the badge as well.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/14044314/3566f375-69f6-4748-a962-c347845c9e52



## Note:
You need to click "Save Changes" on the badge modal to save deletion, otherwise, it will revert back to the original server-saved badge count.
![Save_Changes_Button](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/14044314/5efeabdd-049e-4818-93b2-e7e3998e5c47)
